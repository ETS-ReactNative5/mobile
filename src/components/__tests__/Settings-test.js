import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Settings } from '../Settings'

jest.mock('Switch', () => 'Switch');

const notifications = {
  'general': true
}

const settings = {
  'promptForWorkflow': true
}

it('renders', () => {
  const tree = renderer.create(
    <Settings
      notifications={notifications}
      checkPushPermissions={jest.fn}
      pushEnabled={true}
      settings={settings}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders message if push is disabled', () => {
  const tree = renderer.create(
    <Settings
      notifications={notifications}
      checkPushPermissions={jest.fn}
      pushEnabled={false}
      settings={settings}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
