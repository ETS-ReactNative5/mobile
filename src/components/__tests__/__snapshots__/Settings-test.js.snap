// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <RCTScrollView>
    <View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          General Settings
        </Text>
        <View
          style={undefined}
        >
          <Switch
            onTintColor={undefined}
            onValueChange={[Function]}
            style={undefined}
            value={true}
          />
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                ]
              }
            >
              Prompt to see all project workflows
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              Note: This will include non-mobile workflows
            </Text>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        Notification Settings
      </Text>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
            ]
          }
        >
          Zooniverse would like to occassionally send you updates about new projects or projects needing help.
        </Text>
        <View
          style={undefined}
        >
          <Switch
            onTintColor={undefined}
            onValueChange={[Function]}
            style={undefined}
            value={true}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }
          >
            General Zooniverse notifications
          </Text>
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            Project-specific Notifications
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders message if push is disabled 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <RCTScrollView>
    <View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          General Settings
        </Text>
        <View
          style={undefined}
        >
          <Switch
            onTintColor={undefined}
            onValueChange={[Function]}
            style={undefined}
            value={true}
          />
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                ]
              }
            >
              Prompt to see all project workflows
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              Note: This will include non-mobile workflows
            </Text>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        Notification Settings
      </Text>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
            ]
          }
        >
          Push notifications are not enabled on your device!  Please go to Settings &gt; Notifications &gt; Zooniverse to allow them.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
